presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestJdbcConnection.java

```diff
@@ -247,6 +257,32 @@ public void testExtraCredentials()
         assertEquals(listExtraCredentials(connection), credentials);
     }
 
+    @Test
+    public void testCustomHeaders()
+            throws SQLException, UnsupportedEncodingException
+    {
+        Map<String, String> customHeadersMap = ImmutableMap.of("testHeaderKey", "testHeaderValue");
+        String customHeaders = "{'testHeaderKey':'testHeaderValue'}";
+        String encodedCustomHeaders = URLEncoder.encode(customHeaders, StandardCharsets.UTF_8.toString());
+        Connection connection = createConnection("customHeaders=" + encodedCustomHeaders);
+        assertTrue(connection instanceof PrestoConnection);
+        PrestoConnection prestoConnection = connection.unwrap(PrestoConnection.class);
+        assertEquals(prestoConnection.getCustomHeaders(), customHeadersMap);
+    }
+
+    @Test(dataProvider = "customHeaderWithSpecialCharacter")
+    public void testCustomHeadersWithSpecialCharacters(String header)
```

Comment:
maybe rename the arg as headerValue so that is is clear

Reply:
✅